userspace/libsinsp/filterchecks.cpp

```diff
@@ -109,6 +110,317 @@ static inline bool str_match_start(const std::string& val, size_t len, const cha
+		{PT_CHARBUF, EPF_NONE, PF_NA, "fspath.sourceraw", "Source path for Filesystem-related operation", "For any event type that deals with a filesystem path, and specifically for a source and target like mv, cp, etc, the source path the file syscall is operating on. This path is always the path provided to the syscall and may not be fully resolved."},
+		{PT_CHARBUF, EPF_NONE, PF_NA, "fspath.target", "Target path for Filesystem-related operation", "For any event type that deals with a filesystem path, and specifically for a target and target like mv, cp, etc, the target path the file syscall is operating on. This path is always fully resolved, prepending the thread cwd when needed."},
+		{PT_CHARBUF, EPF_NONE, PF_NA, "fspath.targetraw", "Target path for Filesystem-related operation", "For any event type that deals with a filesystem path, and specifically for a target and target like mv, cp, etc, the target path the file syscall is operating on. This path is always the path provided to the syscall and may not be fully resolved."},
+};
+
+sinsp_filter_check_fspath::sinsp_filter_check_fspath()
+{
+	m_info.m_name = "file";
+	m_info.m_desc = "Every syscall that has a filesystem path in its arguments has these fields set with information related to the path arguments. This differs from the fd.* fields as it includes syscalls like unlink, rename, etc. that act directly on filesystem paths as compared to opened file descriptors.";
+	m_info.m_fields = sinsp_filter_check_fspath_fields;
+	m_info.m_nfields = sizeof(sinsp_filter_check_fspath_fields) / sizeof(sinsp_filter_check_fspath_fields[0]);
+	m_info.m_flags = filter_check_info::FL_WORKS_ON_THREAD_TABLE;
+};
+
+std::shared_ptr<gen_event_filter_check> sinsp_filter_check_fspath::create_check(const char *name,
+									      cmpop cop,
+									      const char *value)
+{
+	gen_event_filter_check *chk = m_factory->new_filtercheck(name);
+	chk->m_cmpop = cop;
+	chk->m_boolop = BO_NONE;
+	chk->parse_field_name(name, true, true);
+
+	if(value)
+	{
+		chk->add_filter_value(value, strlen(value), 0);
+	}
+
+	std::shared_ptr<gen_event_filter_check> ret(chk);
+	return ret;
+}
+
+void sinsp_filter_check_fspath::create_fspath_checks()
+{
+	m_factory.reset(new sinsp_filter_factory(m_inspector));
+
+	std::shared_ptr<gen_event_filter_check> evt_arg_path = create_check("evt.rawarg.path");
+	std::shared_ptr<gen_event_filter_check> evt_arg_pathname = create_check("evt.rawarg.pathname");
+	std::shared_ptr<gen_event_filter_check> evt_arg_res_eq_0 = create_check("evt.rawarg.res", CO_EQ, "0");
+	std::shared_ptr<gen_event_filter_check> evt_arg_name = create_check("evt.rawarg.name");
+	std::shared_ptr<gen_event_filter_check> evt_fd_name = create_check("fd.name");
```

Comment:
Preference would be to rather also use the `evt.rawarg*` here to have a clear separation / clean slate between the field classes. Even have the suspicion the old processing for `fd.name` can be cleaned up and simplified in a follow up PR.

Reply:
I could use evt.rawarg for the open variants but not for fchmod/fchown, as they act on a fd but don't have the paths in their own event.